src/mon/OSDMonitor.cc

```diff
@@ -3345,6 +3354,35 @@ void OSDMonitor::take_all_failures(list<MonOpRequestRef>& ls)
   failure_info.clear();
 }
 
+int OSDMonitor::get_grace_interval_threshold()
+{
+  int halflife = g_conf()->mon_osd_laggy_halflife;
+  // Scale the halflife period (default: 1_hr) by
+  // a factor (48 hrs) to calculate the threshold.
+  int grace_threshold_hrs = 48;
+  return halflife * grace_threshold_hrs;
+}
+
+bool OSDMonitor::grace_interval_threshold_exceeded(int last_failed_interval)
+{
+  int grace_interval_threshold_secs = get_grace_interval_threshold();
+  dout(10) << __func__ << " last_failed_interval = " << last_failed_interval
```

Comment:
how about having a message at level 1 when last_failed_interval > grace_interval_threshold_secs?

Reply:
Done. Message is logged when failed interval exceeds the threshold at level 1.